var/spack/repos/builtin/packages/strumpack/package.py

```diff
@@ -59,8 +64,11 @@ class Strumpack(CMakePackage):
     depends_on('parmetis', when='+parmetis')
     depends_on('scotch~metis', when='+scotch')
     depends_on('scotch~metis+mpi', when='+scotch+mpi')
+    depends_on('butterflypack@1.1.0:', when='+butterflypack+mpi')
 
     conflicts('+parmetis', when='~mpi')
+    conflicts('+butterflypack', when='~mpi')
+    conflicts('+butterflypack', when='strumpack@:3.2.0')
```

Comment:
Are we sure we want `+butterflypack` to default to True? That means building any older version will fail unless someone manually changes to `~butterflypack`.

Reply:
+butterflypack is an important new development in strumpack, so we want this to be default.
One might 'forget' to add +butterflypack, resulting in an install with limited usability. But when you forget to add ~butterflypack, you get a useful
  ".1. "+butterflypack" conflicts with "strumpack@:3.2.0"
and you can just add that missing ~butterflypack, before waiting for the install.

Is there a way to set the default value of the variant based on the version?

Reply:
Alright, I'm fine with this explanation.

> Is there a way to set the default value of the variant based on the version?

Not yet unfortunately.